mono/mini/debugger-agent.c

```diff
@@ -9204,7 +9204,7 @@ object_commands (int command, guint8 *p, guint8 *end, Buffer *buf)
 	switch (command) {
 	case CMD_OBJECT_REF_GET_TYPE:
 		/* This handles transparent proxies too */
-		buffer_add_typeid (buf, obj->vtable->domain, mono_class_from_mono_type (((MonoReflectionType*)obj->vtable->type)->type));
+		buffer_add_typeid (buf, obj->vtable->domain, mono_class_from_mono_type (mono_vtable_get_reflection_type (obj->vtable)->type));
```

Comment:
This looks both wrong and wasteful.

First, you're allocating memory from the debugger thread.
Second, if the goal is to get the object type, obj->vtable->klass would have done it great.